packages/blocks/src/api/factory.js

```diff
@@ -212,6 +232,57 @@ const getBlockTypesForPossibleToTransforms = ( blocks ) => {
 	return blockNames.map( ( name ) => getBlockType( name ) );
 };
 
+/**
+ * Determines whether transform is a "block" type
+ * and if so whether it is a "wildcard" transform
+ * ie: targets "any" block type
+ *
+ * @param  {Object} t the Block transform object
+ *
+ * @return {boolean} whether transform is a wildcard transform
+ */
+export const isWildcardBlockTransform = ( t ) => t && t.type === 'block' && t.blocks.length && t.blocks.includes( '*' );
```

Comment:
What does `t.blocks.length &&` accomplish here? My first guess was that it guards against a transform which does not define a `blocks` array but in closer inspection, it's not clear we would _want_ to allow that, and this would provide no such guard anyways, since if `t.blocks` would undefined, the property access `.length` would throw an error.

If we did want to provide guarding here, I'd suggest we both (a) include a test case (since it currently passes even without this condition) and (b) change this to `Array.isArray( t.blocks )`

```suggestion
export const isWildcardBlockTransform = ( t ) => t && t.type === 'block' && Array.isArray( t.blocks ) && t.blocks.includes( '*' );
```

Alternatively, the [Lodash variant of `includes`](https://lodash.com/docs/4.17.11#includes) has falsey consideration baked-in.

```
⚡ ⇒ n_
n_ > _.includes( undefined, '*' )
false
```

Reply:
Agreed. I was incorrectly using `.length` to test for an array. I've correct as you suggest. I'll try to be mindful of this approach in future.